src/main/java/net/minecraftforge/fluids/FluidDictionary.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.fluids;
+
+import com.google.common.collect.Maps;
+import java.util.Map;
+import javax.annotation.Nullable;
+import net.minecraft.util.NonNullList;
+
+public class FluidDictionary
+{
+    private static final Map<Fluid, NonNullList<String>> FLUID_TO_NAMES = Maps.<Fluid, NonNullList<String>>newHashMap();
+    private static final Map<String, NonNullList<Fluid>> NAME_TO_FLUIDS =  Maps.<String, NonNullList<Fluid>>newHashMap();
+    
+    public static void registerFluid(Fluid fluid, String name)
+    {
+        name = name.toLowerCase();
```

Comment:
Use `com.google.common.base.Preconditions#checkArgument` at the top of the method here to make sure the fluid is in the fluid registry first, it will prevent some confusion.